metrics/wait.io.table/table.go

```diff
@@ -0,0 +1,236 @@
+// Copyright 2022 Block, Inc.
+
+package waitiotable
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	"github.com/cashapp/blip"
+)
+
+const (
+	DOMAIN = "wait.io.table"
+
+	OPT_EXCLUDE        = "exclude"
+	OPT_INCLUDE        = "include"
+	OPT_TRUNCATE_TABLE = "truncate-table"
+	OPT_ALL            = "all"
+
+	TRUNCATE_QUERY = "TRUNCATE TABLE performance_schema.table_io_waits_summary_by_table"
+)
+
+var (
+	metric_names = []string{
+		"sum_timer_wait",
+		"min_timer_wait",
+		"avg_timer_wait",
+		"max_timer_wait",
+		"count_read",
+		"sum_timer_read",
+		"min_timer_read",
+		"avg_timer_read",
+		"max_timer_read",
+		"count_write",
+		"sum_timer_write",
+		"min_timer_write",
+		"avg_timer_write",
+		"max_timer_write",
+		"count_fetch",
+		"sum_timer_fetch",
+		"min_timer_fetch",
+		"avg_timer_fetch",
+		"max_timer_fetch",
+		"count_insert",
+		"sum_timer_insert",
+		"min_timer_insert",
+		"avg_timer_insert",
+		"max_timer_insert",
+		"count_update",
+		"sum_timer_update",
+		"min_timer_update",
+		"avg_timer_update",
+		"max_timer_update",
+		"count_delete",
+		"sum_timer_delete",
+		"min_timer_delete",
+		"avg_timer_delete",
+		"max_timer_delete",
+	}
+
+	metric_map map[string]struct{}
+)
+
+func init() {
+	metric_map = make(map[string]struct{}, len(metric_names))
+	for _, name := range metric_names {
+		metric_map[name] = struct{}{}
+	}
+}
+
+type tableOptions struct {
+	query    string
+	truncate bool
+}
+
+// Table collects table io for domain wait.io.table.
+type Table struct {
+	db *sql.DB
+	// --
+	options map[string]*tableOptions
+}
+
+// Verify collector implements blip.Collector interface.
+var _ blip.Collector = &Table{}
+
+// NewTable makes a new Table collector,
+func NewTable(db *sql.DB) *Table {
+	return &Table{
+		db:      db,
+		options: map[string]*tableOptions{},
+	}
+}
+
+// Domain returns the Blip metric domain name (DOMAIN const).
+func (t *Table) Domain() string {
+	return DOMAIN
+}
+
+// Help returns the output for blip --print-domains.
+func (t *Table) Help() blip.CollectorHelp {
+	return blip.CollectorHelp{
+		Domain:      DOMAIN,
+		Description: "Table IO Waits",
+		Options: map[string]blip.CollectorHelpOption{
+			OPT_INCLUDE: {
+				Name: OPT_INCLUDE,
+				Desc: "Comma-separated list of database or table names to include (overrides option " + OPT_EXCLUDE + ")",
+			},
+			OPT_EXCLUDE: {
+				Name:    OPT_EXCLUDE,
+				Desc:    "Comma-separated list of database or table names to exclude (ignored if " + OPT_INCLUDE + " is set)",
+				Default: "mysql.*,information_schema.*,performance_schema.*,sys.*",
```

Comment:
Let's put this string in a `const` because it's used below in
```go
dom.Options[OPT_EXCLUDE] = "mysql.*,information_schema.*,performance_schema.*,sys.*"
```
Without a single `const` to keep this DRY, some day some future eng will change one but not the other.

Reply:
Do you want that to be a const in just this metric? We use this across multiple metrics.

Reply:
Yeah, I've noticed this whole idea of include/exclude tables is popping up in several places. We can leave this here as-is for now, and later we can factor out the common stuff somewhere/somehow.